internal/observability.go

```diff
@@ -65,42 +67,111 @@ var ViewExportedSpans = &view.View{
 	TagKeys:     []tag.Key{tagKeyExporterName},
 }
 
+var mLatencyMs = stats.Float64("oc.io/latency", "The latency of the various actions in milliseconds", "ms")
+
+var latencyDistribution = view.Distribution(
+	0, 0.01, 0.05, 0.1, 0.3, 0.6, 0.8, 1, 2, 3, 4, 5, 6, 8, 10, 13, 16, 20, 25, 30, 40, 50, 65, 80,
+	100, 130, 160, 200, 250, 300, 400, 500, 650, 800, 1000, 2000, 5000, 10000, 20000, 50000, 100000,
+)
+
+// VIewLatency defines the view to extract the latencies for various actions.
+var ViewLatency = &view.View{
+	Name:        mLatencyMs.Name(),
+	Description: mLatencyMs.Description(),
+	Measure:     mLatencyMs,
+	Aggregation: latencyDistribution,
+	TagKeys:     []tag.Key{tagKeyReceiverName, tagKeyExporterName},
+}
+
 // AllViews has the views for the metrics provided by the agent.
 var AllViews = []*view.View{
 	ViewReceivedSpansReceiver,
 	ViewExportedSpans,
+	ViewLatency,
+}
+
+// NewReceiverEventRecorder creates a helper function that'll add the name of the
+// creating exporter as a tag value in the context that will be used to count the
+// the number of spans exported, but it will also create a span to mark the duration
+// of activeness of the caller.
+func NewReceiverEventRecorder(receiverName string) SpansObservabilityRecorder {
+	return &spansRecorder{
+		predefinedSpanName: "opencensus.service.receiver." + receiverName + ".Receive",
+		startTime:          time.Now(),
+		receiverName:       receiverName,
+	}
+}
+
+// SpansObservabilityRecorder is a helper that is started when
+type SpansObservabilityRecorder interface {
+	Start(ctx context.Context, ni *commonpb.Node) context.Context
+	End(ctx context.Context, nItems int64, err error) context.Context
+	UnderlyingSpan() *trace.Span
+}
+
+type spansRecorder struct {
+	predefinedSpanName string
+	span               *trace.Span
+	receiverName       string
+	exporterName       string
```

Comment:
I think you can form the wanted tags on construction (so they don't need to be reconstructed frequently), and then get rid of any receiver/exporter specificity within this struct itself